### DISCUSSION TITLE
Can mergify be told to add the original authors of strict rebased pr's as Co-Authors?

### DISCUSSION THREAD

**Discussion by AraHaan** [Mergifyio/mergify#2468] [Q&A] [answered]
I would like to now if Mergify can be told to add the original authors of strict rebased pr's as Co-Authors if it changes the author of the commits inside?

The docs seems to not at least list an option to at least tell mergify that you want to set the original author as a co-author in the rebased commits on the PR if strict rebase is set.

And yes I decided to migrate from rebase only PRs from the automerge github actions (which I had to hardcode users on the workflows which is not so nice really instead of hardcode it on just the label named ``automerge`` then use mergify).

As such I done this sort of merge options set:
- strict: smart+fasttrack
- strict_method: rebase
- method: squash

As such I actually prefer the original authors to at least remain on the commits somewhere at least so I can properly ``git blame`` the right person since I cannot afford to upgrade from free plan for now (at least until I figure out setting up project donations on github).

**Comment by sileht** [comment, marked as answer]
The original author is always accessible, mergify[bot] is only the committer, not the author, for merge commits, and update commits.
For squash commit (built by GitHub and not Mergify) GitHub adds by default to the message: `Co-authored-by: ...`

Otherwise, this can be done with `update_bot_account: "{{ author }}"` and `merge_bot_account: "{{ author }}"`, but the author needs to be parts of the organization and have logged at least once on the Dashboard.

**Reply by AraHaan** [reply]
So basically for the rebase the bot is the committer, and does not take ownership of the actual commit itself (it just rebased and pushed it all and kept the original author data intact.

Also I later realized I can replace the action/add-labels and action/remove-labels with the bot's config file rules. It really is simplifying my github actions workflows quite a lot from hard coding them in. However I would need to check the opener for if they are a part of the organization however before adding the automerge label through the pr rules to avoid automerging prs from forks without manual intervention and an actual human adding the automerge label to trigger the bot to merge.

Perhaps there is a simple check to check if:
- the user is an outside contributor to the org (is a contributor added directly to the repository itself that the pr is in)
- is a member of the org but not part of a team inside it
- is a member and part of a team.

**Reply by jd** [reply]
We don't expose such info about author yet, but that might be doable in the future. Feel free to open an issue.
